samtranslator/model/sam_resources.py

```diff
@@ -2407,3 +2418,148 @@ def _set_resolver_code_settings(resolver_code_settings: aws_serverless_graphqlap
+                        relative_id, "'Id' cannot be defined with other properties in Function."
+                    )
+                continue
+
+            if not function.InlineCode and not function.CodeUri:
+                raise InvalidResourceException(relative_id, "One of 'InlineCode' or 'CodeUri' must be set.")
+
+            if function.InlineCode and function.CodeUri:
+                raise InvalidResourceException(
+                    relative_id, "Both 'InlineCode' and 'CodeUri' cannot be defined at the same time."
+                )
+
+            if function.InlineCode and not function.Runtime:
+                raise InvalidResourceException(
+                    relative_id,
+                    "'Runtime' must be defined when 'InlineCode' is defined.",
+                )
+
+            if not function.DataSource and not function.DataSourceName:
+                raise InvalidResourceException(relative_id, "One of 'DataSource' or 'DataSourceName' must be set.")
+
+            if function.DataSource and function.DataSourceName:
+                raise InvalidResourceException(
+                    relative_id, "Both 'DataSourceType' and 'DataSourceName' cannot be defined at the same time."
+                )
+
+            func_config.Code = function.InlineCode
+            func_config.CodeS3Location = function.CodeUri
+            func_config.ApiId = api_id
+            func_config.Name = function.Name or relative_id
+            func_config.DataSourceName = self._parse_datasource_name(relative_id, function, datasources)
+            func_config.MaxBatchSize = cast(PassThrough, function.MaxBatchSize)
+            func_config.Description = cast(PassThrough, function.Description)
+
+            if function.Sync:
+                func_config.SyncConfig = self._parse_function_sync(function.Sync)
+
+            if function.Runtime:
+                func_config.Runtime = {
+                    "Name": cast(PassThrough, function.Runtime.Name),
+                    "RuntimeVersion": cast(PassThrough, function.Runtime.Version),
+                }
+
+            resources.append(func_config)
+
+        return resources
+
+    def _check_and_construct_none_datasource(
+        self, functions: Dict[str, aws_serverless_graphqlapi.Function], api_id: Intrinsicable[str]
+    ) -> Optional[DataSource]:
+        """
+        Checks if a DataSource with type "NONE" needs to be created.
+
+        Within a Serverless::GraphQLApi resource, we only generate one "NoneDataSource"
+        that functions can use. This function checks to see if the customer is using a
+        NoneDataSource and constructs it and returns. If there is no functions that use
+        a NoneDataSource, then we return None.
+        """
+        for function in list(functions.values()):
+            if function.Name == "NONE":
+                logical_id = f"{self.logical_id}NoneDataSource"
+
+                none_datasource = DataSource(
```

Comment:
Do we actually check that there hasn't existed any `NoneDataSource` before?

Reply:
We only create one `NoneDataSource` in this function. It immediately returns after